Project-Aurora/Project-Aurora/Devices/UnifiedHID/RoccatTyon.cs

```diff
@@ -0,0 +1,153 @@
+                {
+                    if (buffer[1] == 0x01)
+                        return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            return false;
+        }
+
+        public override bool Connect()
+        {
+            if (!Global.Configuration.VarRegistry.GetVariable<bool>($"UnifiedHID_{this.GetType().Name}_enable"))
+            {
+                return false;
+            }
+            IEnumerable<HidDevice> devices = HidDevices.Enumerate(0x1E7D, new int[] { 0x2E4A });
+            try
+            {
+                if (devices.Count() > 0)
+                {
+                    ctrl_device_leds = devices.First(dev => dev.Capabilities.UsagePage == 0x0001 && dev.Capabilities.Usage == 0x0002);
+                    ctrl_device = devices.First(dev => dev.Capabilities.FeatureReportByteLength > 50);
+
+                    ctrl_device.OpenDevice();
+                    ctrl_device_leds.OpenDevice();
+                    bool success = InitMouseColor() && WaitCtrlDevice();
+                    if (!success)
+                    {
+                        Global.logger.LogLine($"Roccat Tyon Could not connect\n", Logging_Level.Error);
+                        ctrl_device.CloseDevice();
+                        ctrl_device_leds.CloseDevice();
+                    }
+                    Global.logger.LogLine($"Roccat Tyon Connected\n", Logging_Level.Info);
+                    return (IsConnected = success);
+                }
+            }
+            catch (Exception exc)
+            {
+                Global.logger.LogLine($"Error when attempting to open UnifiedHID device:\n{exc}", Logging_Level.Error);
+            }
+            return false;
+        }
+
+        // We need to override Disconnect() too cause we have two HID devices open for this keyboard.
+        public override bool Disconnect()
+        {
+            try
+            {
+                ctrl_device.CloseDevice();
+                ctrl_device_leds.CloseDevice();
+                return true;
+            }
+            catch (Exception exc)
+            {
+                Global.logger.LogLine($"Error when attempting to close UnifiedHID device:\n{exc}", Logging_Level.Error);
+            }
+            return false;
+        }
+
+        public override bool SetLEDColour(DeviceKeys key, byte red, byte green, byte blue)
+        {
+            try
+            {
+                if (!this.IsConnected)
+                    return false;
+                byte[] hwmap = new byte[11];
+
+                hwmap[0] = red;
+                hwmap[1] = green;
+                hwmap[2] = blue;
+                hwmap[3] = 0x00;
+                hwmap[4] = 0x00;
+                hwmap[5] = red;
+                hwmap[6] = green;
+                hwmap[7] = blue;
+                hwmap[8] = 0x00;
+                hwmap[9] = 0x80;
+                hwmap[10] = 0x80;
```

Comment:
Could save some lines and make it look nicer by putting these in the initialisation of the byte array rather than assigning the values to each index

Reply:
Done did, thanks.